src/common/dpar_mpi.c

```diff
@@ -0,0 +1,350 @@
+#include <stdio.h>
+#include <mpi.h>
+#include <daos/dpar.h>
+#include <gurt/atomic.h>
+
+static ATOMIC uint64_t	comm_free_bits = 0xfffffffffffffffe;
+static MPI_Comm		comm_table[64];
```

Comment:
MPI_Comm is not the same size between openmpi and mpich.  Didn't check Intel MPI.  Anyway, I figured that 64 communicators should be enough given we use only PAR_COMM_WORLD today.

Reply:
I think intel MPI is the same as mpich in that regards. anyway thanks for adding that. was coming to approve this and saw you added it :-)